packages/remix-server-runtime/server.ts

```diff
@@ -400,7 +400,15 @@ async function handleDocumentRequestRR(
       };
       break;
     } else {
-      appState.loaderBoundaryRouteId = id;
+      // Only assign the boundary id if this module has a boundary.  This
+      // should be true in almost all cases, except for cases in which no
+      // boundaries exist and the router "assigns" the error to the root route
+      // for lack of a better place to put it.  If the root doesn't have an
+      // error boundary, then we leave loaderBoundaryId null to bubble to the
+      // default boundary at render time
+      if (build.routes[id]?.module.ErrorBoundary) {
+        appState.loaderBoundaryRouteId = id;
+      }
```

Comment:
@jacob-ebey I wouldn't mind talking through the current router approach with you tomorrow for not-founds and error bubbling up _through _ the root to the Remix-provided defaults.  Might be something we can clean up.